### PR TITLE
Feature: add activecampaign add-on to the form migration process

### PR DESCRIPTION
**Pull request by JoshuaHungDinh**
Resolves # [GIVE-452]

## Description

This PR adds a step in the form migration process to convert ActiveCampaign data from v2 forms to v3 forms and includes the methods required to facilitate this in the FormMetaDecorator. 

It also includes unit tests on the form migration process for the ActiveCampaign add-on. Three tests were added that check when migrating a form that uses the global settings, migrating a form that has per-form settings & migrating a form without meta.

## Additional context
While Migrating the form if enabled globally from the global settings is selected the form migration should assume that it should migrate the form with the add-on block. On the v2 form if "use default" is selected the form should migrate with global settings. If "customize" is selected it should migrate via the form settings. If "disabled" is selected while enabled globally is selected it should not migrate the form with the add-on block. 

## Affects
Form Migration

## Testing Instructions
- Set your global settings
- add a v2 form -> then migrate the form without editing the form settings.
- Verify the v3 form has the correct settings set from the global options.
- Create another v2 form -> update the per-form settings on the right -> then migrate the form
- Verify the v3 form has the correct settings set from the form options.

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-452]: https://stellarwp.atlassian.net/browse/GIVE-452?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/FormMigration/FormMetaDecorator.php [hunk 1]
```diff
@@ -784,4 +784,53 @@ public function getFormFeaturedImage()
...
         return $featuredImage;
     }
+
+    /**
+     * @unreleased
+     */
+    public function isActiveCampaignEnabled(): bool
+    {
+        return true;
+    }
```
Comment (jonwaldstein) [inline src/FormMigration/FormMetaDecorator.php:794]: @JoshuaHungDinh this doesn't look right lol we want to actually check if it's enabled

**Review by jonwaldstein — CHANGES_REQUESTED** [review]

File: src/FormMigration/FormMetaDecorator.php [hunk 2]
```diff
@@ -784,4 +785,71 @@ public function getFormFeaturedImage()
...
+        $isFormDisabled = give_is_setting_enabled($this->getMeta('activecampaign_per_form_options'), 'disabled');
+
+        $isGloballyEnabled = give_is_setting_enabled(give_get_option('give_activecampaign_globally_enabled'), 'on');
+
+        Log::error('isActiveCampaignEnabled', [
+            'return' => !($isFormDisabled || (!$isGloballyEnabled && !$isFormEnabled)),
+            '$isFormEnabled' =>$isFormEnabled,
+            '$isFormDisabled' => $isFormDisabled,
+            '$isGloballyEnabled' => $isGloballyEnabled,
+        ]);
```
Comment (jonwaldstein) [inline src/FormMigration/FormMetaDecorator.php:805]: What's this for?

Comment (JoshuaHungDinh) [reply src/FormMigration/FormMetaDecorator.php]: I was testing the accuracy of the enabled method. It's no longer needed. a1b5fcc69322be0eb3bd94217e446887a4b928e3

**Review by jonwaldstein — APPROVED** [review]
@JoshuaHungDinh great job! thanks for the unit tests too 🦾

**Review by rickalday — APPROVED** [review]
Passed manual QA tests.

**Pull request merged**